### PR TITLE
Improve foreign key description in guides

### PR DESCRIPTION
**Pull request by maicher**
There is an inaccuracy between description and code example in guides.
I propose a code example fix.

### CODE REVIEW

**Comment by robin850** [comment]
Hi @maicher,

Thanks for the pull request ! :-)

Actually, I may be wrong but certainly both `add_foreign_key` and `add_index` should be done. According to [the PostgreSQL guide](https://www.postgresql.org/docs/9.6/static/ddl-constraints.html) on constraints:

> Because this is not always needed, and there are many choices available on how to index, declaration of a foreign key constraint does not automatically create an index on the referencing columns.

However, the index would be still needed for the "performance boost" it could bring if the field is often referenced.

I will let someone which has better database understanding than me merge this. 🙄

**Comment by maicher** [comment]
Hi @robin850, thx for your response.

Foreign keys are for keeping referential data integrity.
Indexes are for the performance boost.

Maybe we could add this to guides? I've just committed my suggestion.

File: guides/source/association_basics.md [hunk 1]
```diff
@@ -562,9 +562,9 @@ You are not free to use just any name for your associations. Because creating an
 
 Associations are extremely useful, but they are not magic. You are responsible for maintaining your database schema to match your associations. In practice, this means two things, depending on what sort of associations you are creating. For `belongs_to` associations you need to create foreign keys, and for `has_and_belongs_to_many` associations you need to create the appropriate join table.
 
-#### Creating Foreign Keys for `belongs_to` Associations
+#### Creating Foreign Keys and Indexes for `belongs_to` Associations
 
-When you declare a `belongs_to` association, you need to create foreign keys as appropriate. For example, consider this model:
+When you declare a `belongs_to` association, It's a good practice to add appropriate foreign keys to ensure referential data integrity and indexes to improve queries performance. For example, consider this model:
```
Comment (Erol) [inline guides/source/association_basics.md]: A couple of suggestions:

* Change _It's_ to _it's_.
* Move the statement about adding indexes to a separate block and give an example where they might be beneficial.

Comment (maicher) [inline guides/source/association_basics.md]: @Erol By moving to a separate block, do you mean creating another titled paragraph, so we will have: "Creating Foreign Keys for `belongs_to` Associations" and "Creating Indexes for `belongs_to` Associations"?

File: guides/source/association_basics.md [hunk 2]
```diff
@@ -562,9 +562,9 @@ You are not free to use just any name for your associations. Because creating an
 
 Associations are extremely useful, but they are not magic. You are responsible for maintaining your database schema to match your associations. In practice, this means two things, depending on what sort of associations you are creating. For `belongs_to` associations you need to create foreign keys, and for `has_and_belongs_to_many` associations you need to create the appropriate join table.
 
-#### Creating Foreign Keys for `belongs_to` Associations
+#### Creating Foreign Keys and Indexes for `belongs_to` Associations
 
-When you declare a `belongs_to` association, you need to create foreign keys as appropriate. For example, consider this model:
+When you declare a `belongs_to` association, it's a good practice to add appropriate foreign keys to ensure referential data integrity and indexes to improve queries performance. For example, consider this model:
```
Comment (vipulnsward) [inline guides/source/association_basics.md]: > it's a good practice

You do need to create the key column, its not optional. Adding the constraint is a good practise.

Can you rephrase this part, and squash the commits.

Comment (maicher) [inline guides/source/association_basics.md]: Hi @vipulnsward 
I've just rephrased it, add example and squash the commits. I believe now it's clearer, however I don't know if informations about foreign key constraint will not be confusing in that place.

File: guides/source/association_basics.md [hunk 3]
```diff
@@ -582,14 +582,29 @@ class CreateBooks < ActiveRecord::Migration[5.0]
...
     end
-
-    add_index :books, :author_id
   end
 end
 ```
 
 If you create an association some time after you build the underlying model, you need to remember to create an `add_column` migration to provide the necessary foreign key.
 
+It's a good practice to add an index on the foreign key to improve queries performance and a foreign key constraint to ensure referential data integrity:
```
Comment (robin850) [inline guides/source/association_basics.md]: Could you wrap this line around 80 chars please ?

Comment (maicher) [inline guides/source/association_basics.md]: Done.

**Comment by robin850** [comment]
Thanks ! :-)
